Libplanet/Blocks/Block.cs

```diff
@@ -97,10 +96,10 @@ private Block(RawBlock rb)
         public DateTimeOffset Timestamp { get; }
 
         [IgnoreDuringEquals]
-        public IEnumerable<Transaction<T>> Transactions { get; }
+        public Transaction<T>[] Transactions { get; }
```

Comment:
As we've encouraged immutable data structures over mutable ones (in particular for model types), it would be better if it uses [`ImmutableArray<T>`][1] instead:

```suggestion
        public ImmutableArray<Transaction<T>> Transactions { get; }
```

Of course, this would bring the changes to the constructor as well:

```diff
diff --git a/Libplanet/Blocks/Block.cs b/Libplanet/Blocks/Block.cs
index 1651ad0e..63fbc66c 100644
--- a/Libplanet/Blocks/Block.cs
+++ b/Libplanet/Blocks/Block.cs
@@ -40,14 +40,14 @@ namespace Libplanet.Blocks
             Miner = miner;
             PreviousHash = previousHash;
             Timestamp = timestamp;
-            Transactions = transactions.ToArray();
+            Transactions = transactions.ToImmutableArray();
             Hash = Hashcash.Hash(ToBencodex(false, false));

             var hashInteger = new BigInteger(Hash.ToByteArray());
             OrderedTransactions =
                 Transactions
                     .OrderBy(tx =>
-                        new BigInteger(tx.Id.ToByteArray()) ^ hashInteger).ToArray();
+                        new BigInteger(tx.Id.ToByteArray()) ^ hashInteger).ToImmutableArray();
         }

         protected Block(SerializationInfo info, StreamingContext context)
```

[1]: ﻿https://devblogs.microsoft.com/dotnet/please-welcome-immutablearrayt/